### PR TITLE
Inorder traversal added

### PR DESCRIPTION
**Pull request by shivammahajancse**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Yes)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(#177 )

### CODE REVIEW

**Comment by shivammahajancse** [comment]
Please review sir @HarshCasper @atarax665 @ankitaggarwal23 @gat786

**Comment by shivammahajancse** [comment]
Sir @iamrajiv please review

**Comment by nidh-ish** [comment]
@shivammahajancse Resolve the issue of the conflicting files.

File: docs/data-structures/inordertraversal.mdx [hunk 1]
```diff
@@ -0,0 +1,206 @@
+---
+title: Inorder traversal of Binary Tree
+slug: /data-structures/inordertraversal
+---
+
+## Introduction
+
+Binary Tree is a type of tree which have atmost two children or we can say atmost degree 2. So to traverse the binary tree according to Depth first search. We have a traversal algorithm called Inorder traversal.
```
Comment (nidh-ish) [inline docs/data-structures/inordertraversal.mdx:8]: Few grammatical mistakes over here. Make it:

A binary tree is a type of tree which have at most two children or we can say at most degree 2. So to traverse the binary tree according to Depth-first search. We have a traversal algorithm called Inorder traversal.

File: docs/data-structures/inordertraversal.mdx [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+
+## Explanation 
+In this type of traversal(Inorder traversal). First we have to traverse the left part then its root then right part respectively. In left part we have to traverse whole left subtree after that root of thet subtree and then whole right subtree. Inorder: (left->root->right).
```
Comment (nidh-ish) [inline docs/data-structures/inordertraversal.mdx:11]: Make it:

In this type of traversal(Inorder traversal), first, we have to traverse the left part then its root then the right part respectively. In the left part, we have to traverse the whole left subtree after that root of that subtree and then the whole right subtree. Inorder: (left->root->right).

File: docs/data-structures/inordertraversal.mdx [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+For Inorder traversal, we follow the following algorithm:
+
+* First we have to traverse the Binary tree in this order(left->root->right).
+
+* Left means traverse the whole left subtree until we get left child as null. 
+
+* Root means parent if that particular subtree.
+
+* Right means traverse the whole right subtree in this (left->root->right) order until we get null.
```
Comment (nidh-ish) [inline docs/data-structures/inordertraversal.mdx:47]: Make it:

* First, we have to traverse the Binary tree in this order(left->root->right).
* Left means traversing the whole left subtree until we get the left child as null. 
* Root means the parent of that particular subtree.
* Right means traverse the whole right subtree in this (left->root->right) order until we get null.

File: docs/data-structures/inordertraversal.mdx [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+  i+=1
+print('In-order traversal : ')
+# In-Order Traversal :- left -> root -> right
+inorder(tree)
+    
+```
+</TabItem>
+</Tabs>
+
+# Sample Input and Output
```
Comment (nidh-ish) [inline docs/data-structures/inordertraversal.mdx:182]: Use heading 2

File: docs/data-structures/inordertraversal.mdx [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+
+
+#### Input :
+
+```
+Enter the data: 44 22 66 11 33 55 77 
+
```
Comment (nidh-ish) [inline docs/data-structures/inordertraversal.mdx:189]: Remove this extra line

File: docs/data-structures/inordertraversal.mdx [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+In-order traversal : 11 22 33 44 55 66 77
+```
+
+# Time & Space Complexity
+ 
+- **Time Complexity:** `O(n)`
+
+- **Space Complexity:** `O(h)`
+
+where h is the height of binary tree and n is the number of elements.
```
Comment (nidh-ish) [inline docs/data-structures/inordertraversal.mdx:203]: keep h and n in backticks.

File: docs/data-structures/inordertraversal.mdx [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+
+* First we have to traverse the Binary tree in this order(left->root->right).
+
+* Left means traverse the whole left subtree until we get left child as null. 
+
+* Root means parent if that particular subtree.
+
+* Right means traverse the whole right subtree in this (left->root->right) order until we get null.
+
+# Code
```
Comment (nidh-ish) [inline docs/data-structures/inordertraversal.mdx:49]: Use heading 2. Also, the sample I/O for CPP and python doesn't match. Make them uniform across both codes.

File: docs/data-structures/inordertraversal.mdx [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+ Right child of this root: 66
+ It has left child: 55
+ Now its left child is Null
+ so Inorder traversal: 11,22,33,44,55
+ Root of this subtree: 66
+ Right child of this root: 77
+ Now our Inorder traversal: 11,22,33,44,55,66,77 !!
+
+```
+# Algorithm
```
Comment (nidh-ish) [inline docs/data-structures/inordertraversal.mdx:37]: Use heading 2.

File: docs/data-structures/inordertraversal.mdx [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+slug: /data-structures/inordertraversal
+---
+
+## Introduction
+
+Binary Tree is a type of tree which have atmost two children or we can say atmost degree 2. So to traverse the binary tree according to Depth first search. We have a traversal algorithm called Inorder traversal.
+
+## Explanation 
+In this type of traversal(Inorder traversal). First we have to traverse the left part then its root then right part respectively. In left part we have to traverse whole left subtree after that root of thet subtree and then whole right subtree. Inorder: (left->root->right). 
+### Example 1
```
Comment (nidh-ish) [inline docs/data-structures/inordertraversal.mdx:12]: Use heading 4.

File: docs/data-structures/inordertraversal.mdx (hunk 4 above)
Comment (nidh-ish) [inline docs/data-structures/inordertraversal.mdx:182]: Make the sample I/O the same as that is used in the code.

**Review by nidh-ish — CHANGES_REQUESTED** [review]

**Comment by shivammahajancse** [comment]
All requested changes done @iamrajiv @nidh-ish @HarshCasper @ankitaggarwal23

File: docs/data-structures/inordertraversal.mdx [hunk 10]
```diff
@@ -0,0 +1,206 @@
...
+Enter the data: 44 22 66 11 33 55 77 
+
+```
+#### Output :
+
+```
+In-order traversal : 11 22 33 44 55 66 77
+```
+
+## Time & Space Complexity
```
Comment (nidh-ish) [inline docs/data-structures/inordertraversal.mdx:197]: Make it: Complexity Analysis

File: docs/data-structures/inordertraversal.mdx [hunk 11]
```diff
@@ -0,0 +1,206 @@
...
+</TabItem>
+</Tabs>
+
+## Sample Input and Output
+
+#### Input :
+
+```
+Enter the size of tree: 7
+Enter the data: 44 22 66 11 33 55 77
```
Comment (nidh-ish) [inline docs/data-structures/inordertraversal.mdx:188]: Put enter the data as it is asking in the program. Step by step.

**Review by nidh-ish — CHANGES_REQUESTED** [review]
This is the final set of review. Rest looks good.

**Comment by shivammahajancse** [comment]
Changes done @nidh-ish @iamrajiv @ankitaggarwal23 @HarshCasper

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@shivammahajancse Rename the file to `inorder_traversal.mdx`.

**Review by iamrajiv — APPROVED** [review]

**Review by nidh-ish — APPROVED** [review]
LGTM.

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@shivammahajancse Resolve the merge conflict.

**Comment by shivammahajancse** [comment]
Sir done @iamrajiv

**Review by iamrajiv — APPROVED** [review]

**Comment by HarshCasper** [comment]
All the checks have failed. Fix them up.

**Comment by shivammahajancse** [comment]
Sir @HarshCasper done

**Review by atarax665 — APPROVED** [review]

File: sidebars.js [hunk 12]
```diff
@@ -49,9 +49,12 @@ module.exports = {
                 "data-structures/data-structures",
                 "data-structures/binary-tree",
                 "data-structures/Singly-Linked-list",
-                "data-structures/Circular-Linked-list",
-                 "data-structures/binary-search-tree",
+                "data-structures/binary-search-tree",            
+                "data-structures/preorder_traversal",
```
Comment (atarax665) [inline sidebars.js:53]: As this PR is concerned only with ```InorderTraversal``` please remove all other irrelevant changes.

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by shivammahajancse** [comment]
Done @atarax665  sir

File: docs/data-structures/inorder_traversal.mdx [hunk 13]
```diff
@@ -0,0 +1,212 @@
+---
+title: Inorder traversal of Binary Tree
+slug: /data-structures/inorder_traversal
```
Comment (HarshCasper) [inline docs/data-structures/inorder_traversal.mdx:3]: Uff. Do you know how slugs work? You don't insert underscores here

File: docs/data-structures/inorder_traversal.mdx [hunk 14]
```diff
@@ -0,0 +1,212 @@
...
+
+```
+In-order traversal : 11 22 33 44 55 66 77
+```
+
+## Complexity Analysis
+ 
+- **Time Complexity:** `O(n)`
+
+- **Space Complexity:** `O(h)`
```
Comment (HarshCasper) [inline docs/data-structures/inorder_traversal.mdx:207]: Don't use bold unnecessarily

File: docs/data-structures/inorder_traversal.mdx [hunk 15]
```diff
@@ -0,0 +1,212 @@
...
+
+where `h` is the height of binary tree and `n` is the number of elements.
```
Comment (HarshCasper) [inline docs/data-structures/inorder_traversal.mdx:209]: Keep this as a bullet point at the top

File: docs/data-structures/inorder_traversal.mdx [hunk 16]
```diff
@@ -0,0 +1,212 @@
...
+struct Node {
+    int data;
+    struct Node *left;
+    struct Node *right;
+};
+
+// Insert Function
+Node* Insert(Node *root, int data) {
+    //if root have no value
+    //then create a node and insert values
```
Comment (HarshCasper) [inline docs/data-structures/inorder_traversal.mdx:78]: Use Multi-Line Comments

File: docs/data-structures/inorder_traversal.mdx [hunk 17]
```diff
@@ -0,0 +1,212 @@
...
+    if (root == NULL) {
+        root = new Node();
+        root->data = data;
+        root->left = root->right = NULL;
+    }
+    //check if data is less than root value
+    //then insert into left subtree
```
Comment (HarshCasper) [inline docs/data-structures/inorder_traversal.mdx:85]: Use Multi-Line Comments

File: docs/data-structures/inorder_traversal.mdx [hunk 18]
```diff
@@ -0,0 +1,212 @@
...
+    else if (data <= root->data)
+        root->left = Insert(root->left, data);
+    //check if data is less than root value
+    //then insert into right subtree
```
Comment (HarshCasper) [inline docs/data-structures/inorder_traversal.mdx:89]: Use Multi-Line Comments

File: docs/data-structures/inorder_traversal.mdx [hunk 19]
```diff
@@ -0,0 +1,212 @@
...
+ Inorder traversal: 11,22,33,44
+ Right child of this root: 66
+ It has left child: 55
+ Now its left child is Null
+ so Inorder traversal: 11,22,33,44,55
+ Root of this subtree: 66
+ Right child of this root: 77
+ Now our Inorder traversal: 11,22,33,44,55,66,77 !!
+
+```
```
Comment (HarshCasper) [inline docs/data-structures/inorder_traversal.mdx:36]: Use a better example/visualization.

File: docs/data-structures/inorder_traversal.mdx [hunk 20]
```diff
@@ -0,0 +1,212 @@
...
+title: Inorder traversal of Binary Tree
+slug: /data-structures/inorder_traversal
+---
+
+## Introduction
+
+A binary tree is a type of tree which have at most two children or we can say at most degree 2. So to traverse the binary tree according to Depth-first search. We have a traversal algorithm called Inorder traversal.
+
+## Explanation 
+In this type of traversal(Inorder traversal), first, we have to traverse the left part then its root then the right part respectively. In the left part, we have to traverse the whole left subtree after that root of that subtree and then the whole right subtree. Inorder: (left->root->right).
```
Comment (HarshCasper) [inline docs/data-structures/inorder_traversal.mdx:11]: Give a line break and add formatting at apt places

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: docs/data-structures/inorder_traversal.mdx (hunk 13 above)
Comment (shivammahajancse) [reply docs/data-structures/inorder_traversal.mdx:3]: Sir rajiv sir commrnt me to do that

**Comment by shivammahajancse** [comment]
Sir all changes done @HarshCasper

**Pull request merged**